emloop/hooks/classification_metrics.py

```diff
@@ -0,0 +1,87 @@
+"""
+Hook computing epoch statistics for classification tasks.
+"""
+
+from typing import Mapping, List, Union
+import logging
+
+try:
+    import sklearn.metrics as sk
+except ImportError:
+    logging.info('This hook requires SciKit.')
+
+from . import AccumulateVariables
+from ..types import EpochData
+
+
+class ClassificationMetrics(AccumulateVariables):
+    """
+    Accumulate the specified prediction and gt variables and compute their classification statistics after each epoch.
+    In particular, accuracy, precisions, recalls, f1s and sometimes specificity (if f1_average is set to 'binary') are
+    computed and saved to epoch data.
+
+    .. code-block:: yaml
+        :caption: Compute and save classification statistics between model output
+                  `prediction` and stream source `labels`.
+
+        hooks:
+          - ClassificationMetrics:
+              predicted_variable: prediction
+              gt_variable: labels
+    """
+
+    def __init__(self, predicted_variable: str, gt_variable: str, f1_average: str=None, var_prefix: str='', **kwargs):
+        """
+        :param predicted_variable: name of the predicted variable.
+        :param gt_variable: name of the gt variable
+        :param f1_average: averaging type {binary, micro, macro, weighted, samples} defined by
+                           `sklearn.metrics.precision_recall_fscore_support`
+        :param var_prefix: prefix for the output variables
```

Comment:
It would be nice to mention that the prefix should end with a separator, for instance:
```suggestion
        :param var_prefix: prefix for the output variables to avoid name conflicts; e.g. `classification_`
```